extensions/aris/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,493 @@
+					'width': '100%',
+				}
+			});
+
+			allContainer.addItem(passwordInputContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%'
+				}
+			});
+
+			this.rememberPwCheckBox = this.modelBuilder.checkBox()
+				.withProperties<azdata.CheckBoxProperties>({
+					label: 'Remember Password',
+					checked: this.prefilledValues.rememberPassword
+				}).component();
+			this.rememberPwCheckBox.enabled = false;
+
+			allContainer.addItem(this.rememberPwCheckBox, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '0px 0px 0px 65px',
+				}
+			});
+
+			this.errorTextContainer = this.modelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			allContainer.addItem(this.errorTextContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					width: '100%'
+				}
+			});
+
+			tc.addContentContainer(allContainer);
+
+			await view.initializeModel(tc.flexContainer);
+		});
+
+		this.dialog.registerCloseValidator(async () => await this.getEndpoints());
+		this.dialog.cancelButton.onClick(async () => await this.cancel());
+		this.dialog.okButton.label = this.SignInButtonText;
+		this.dialog.cancelButton.label = this.CancelButtonText;
+	}
+
+	private toggleErrorText(errorMessage?: string): void {
+		if (!this.errorTextContainer) {
+			return;
+		}
+
+		this.errorTextContainer.clearItems();
+
+		if (errorMessage) {
+			let errorText = this.modelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({
+					value: errorMessage,
+					CSSStyles: {
+						'color': 'rgb(240, 62, 62)'
+					}
+				})
+				.component();
+
+			this.errorTextContainer.addItem(errorText, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'border': '3px solid rgba(225, 17, 17, 0.5)',
+					'background-color': 'rgba(245, 131, 131, 0.3)',
+					'padding': '0px 16px 0px 16px',
+					'margin-top': '20px',
+					'word-wrap': 'break-word',
+					'width': '100%'
+				}
+			});
+		}
+	}
+
+	private toggleSignInButton(): void {
+		if (!this.dialog || !this.dialog.okButton) {
+			return;
+		}
+		if (this.urlInputBox && this.urlInputBox.value
+			&& this.usernameInputBox && this.usernameInputBox.value
+			&& this.passwordInputBox && this.passwordInputBox.value
+		) {
+			this.dialog.okButton.enabled = true;
+		} else {
+			this.dialog.okButton.enabled = false;
+		}
+	}
+
+	private async getEndpoints(): Promise<boolean> {
+		this.toggleErrorText();
+
+		let url = this.urlInputBox && this.urlInputBox.value;
+		let username = this.usernameInputBox && this.usernameInputBox.value;
+		let password = this.passwordInputBox && this.passwordInputBox.value;
+
+		return await ArisController.getEndPoints(url, username, password, true).then(async response => {
```

Comment:
As mentioned elsewhere, use try/catch and inline the value returned from the await. This massively simplifies the logic.

Reply:
fixed.